base/special/trig.jl

```diff
@@ -80,6 +80,84 @@ end
 sin_kernel(x::Real) = sin(x)
 cos_kernel(x::Real) = cos(x)
 
+# Inverse trigonometric functions
+# atan methods
+ATAN_HI(::Type{Float64}) = (4.63647609000806093515e-01, 7.85398163397448278999e-01,  9.82793723247329054082e-01, 1.57079632679489655800e+00)
+ATAN_HI(::Type{Float32}) = (4.6364760399f-01, 7.8539812565f-01,  9.8279368877f-01, 1.5707962513f+00)
+ATAN_LO(::Type{Float64}) = (2.26987774529616870924e-17,3.06161699786838301793e-17,1.39033110312309984516e-17,6.12323399573676603587e-17)
+ATAN_LO(::Type{Float32}) = (5.0121582440f-09,3.7748947079f-08,3.4473217170f-08,7.5497894159f-08)
+
+ATAN_LARGE_X(::Type{Float64}) = 2.0^66 # seems too large? 2.0^60 gives the same
+ATAN_SMALL_X(::Type{Float64}) = 2.0^-27
+ATAN_LARGE_X(::Type{Float32}) = 2.0f0^26
+ATAN_SMALL_X(::Type{Float32}) = 2.0f0^-12
+
+atan_p(z::Float64, w::Float64) = z*@horner(w,
+     3.33333333333329318027e-01,
+     1.42857142725034663711e-01,
+     9.09088713343650656196e-02,
+     6.66107313738753120669e-02,
+     4.97687799461593236017e-02,
+     1.62858201153657823623e-02)
+atan_q(w::Float64) = w*@horner(w,
+     -1.99999999998764832476e-01,
+     -1.11111104054623557880e-01,
+     -7.69187620504482999495e-02,
+     -5.83357013379057348645e-02,
+     -3.65315727442169155270e-02)
+atan_p(z::Float32, w::Float32) = z*@horner(w, 3.3333328366f-01,  1.4253635705f-01, 6.1687607318f-02)
+atan_q(w::Float32) = w*@horner(w, -1.9999158382f-01, -1.0648017377f-01)
+
+function atan(x::T) where T<:Union{Float32, Float64}
+    absx = abs(x)
+    if absx >= ATAN_LARGE_X(T)
+        if isnan(x)
+              T(NaN)
```

Comment:
Ah, thanks, that was obviously wrong. Forgot it in the int ->float  change of the control flow.